### ISSUE TITLE
android badge counter

### ISSUE THREAD

**Issue by Nick-lab** [phonegap/phonegap-plugin-push#2416]
### Expected Behaviour
when notification comes in with app in background or closed the badge number is set to the number in witch was sent in data>badge = <badge number>
### Actual Behaviour
no badge is set for android.
in data that is received there is a count variable with the correct number that the badge should be.
### Reproduce Scenario (including but not limited to)
Send payload to android device.
#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Android: issue occurs
iOS: everything works perfectly
#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
 Tested on:
 ZTE 7.1.1
ASUS 7.0.0
#### Cordova CLI version and cordova platform version

    cordova --version                                    8.0.0
    cordova platform version android                    7.1.0

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push  2.2.3

#### Sample Push Data Payload
```
{
    notification: {
        sound: 'presto',
        body: 'test message'
    },
    data: {
        message: 'test message',
        badge: 5,
        content-available: 1
    }
}
```
- This push data might be a little bit off from what it fully contains because it is being sent via php server with curl.
#### Logs taken while reproducing problem
``
additionalData
:
coldstart
:
false
content-available
:
"1"
force-start
:
"1"
foreground
:
true
info
:
{notification: "5b1eaff3dd807", checkin: true, user_id: "5b1eafccd447c", push_id: "5b1eb00f7a1a3", message: "Hello Nick, are you OK?"}
no-cache
:
"1"
notId
:
"4197"
__proto__
:
Object
count
:
"1"
message
:
"Hello Nick, are you OK?"
sound
:
"presto"
``

**Comment by Nick-lab** [comment]
tested plugin versions: 
- 2.2.2 - issue persisted
- 2.2.1 - build failed
- 2.2.0 - build failed

attempting @latest Android

**Comment by Nick-lab** [comment]
plugin requires android 7.1.0 and installing plugins versions 2.2.2 and lower generates promise rejection warnings when installing.

**Comment by macdonst** [comment]
@Nick-lab the badge functionality is via [ShortcutBadger](https://github.com/leolin310148/ShortcutBadger). Can you download their test app and see if it works on your phones? Other than that if you update the version of ShortcutBadger to 1.1.21 in your app to see if the latest version fixes your issue.

**Comment by Nick-lab** [comment]
Yes the shortcutBadger test app worked and i use ionic/native badge to alter the badge from in the app.
but how would i go about updating this dependency its not an install in package.json and not in my config.xml. i tried updating the version number in plugins xml file but that didn't seem to change anything. appreciate your help.

**Comment by macdonst** [comment]
@Nick-lab if you clone this repo and then add the plugin from the file system to your project instead of using the one from npm you can make the change.

```
cordova plugin rm phonegap-plugin-push
cordova plugin add ../path/to/phonegap-plugin-push
```

Then build you app it should pull in the latest SB and if that works please send a PR to fix the issue.

**Comment by Nick-lab** [comment]
the latest build does fix the badge count issue. along with clearing the badge when opening the app with pre existing notifications.

**Comment by Nick-lab** [comment]
I couldn't figure out how to get the shortcutBadger dependencies to update in the repo, might be because I haven't messed around with modifying other plugins before. any other recommendations i should try?

**Comment by jcesarmobile** [comment]
It's here https://github.com/phonegap/phonegap-plugin-push/blob/master/plugin.xml#L44

change `1.1.17` to `1.1.22` as it seems it's the latest version

**Comment by Nick-lab** [comment]
and thats it when building will it download the dependencies???

**Comment by jcesarmobile** [comment]
yeah

**Comment by Nick-lab** [comment]
dope trying it now thank you for the clarification.

**Comment by Nick-lab** [comment]
aw man i was hoping this would do the trick but it seems that this is not doing anything. also in the data being sent to android devices i have.
`{ data: {badge: <number>}, notification: {badge: <number> } }` 
not sure if this will conflict with anything.
